File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
input = sys.stdin.readline

N, K = map(int, input().split())

coins = list()

for _ in range(N):
    coins.append(int(input()))

count = 0
for i in range(N):
    coin = coins[N - i - 1]
    if K >= coin:
        count += K // coin
        K %= coin
    if K == 0:
        break

print(count)

File: 백준/Silver/11047. 동전 0/동전 0.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        long K = Long.parseLong(st.nextToken());
        int[] coins = new int[N];
        for (int i = 0; i < N; i++) {
            coins[i] = Integer.parseInt(br.readLine().trim());
        }
        br.close();
        int count = 0;
        for (int i = N - 1; i > -1; i--) {
            int coin = coins[i];
            if(K >= coin){
                count += K / coin;
                K %= coin;
            }
            if(K == 0){
                break;
            }
        }
        System.out.println(count);
    }
}
